### PR TITLE
Swap to attrs

### LINKED ISSUE #16: Investigate Python 3.6 Compatibility
Python 3.6 compatibility would reduce a lot of installation/virtual environment complexity for hosts where Python 3.6 is default. I originally dropped this idea because Python [`dataclasses`](https://docs.python.org/3/library/dataclasses.html), a core piece of the way Aerie CLI handles data, were only introduced in Python 3.7. However, [Pydantic](https://docs.pydantic.dev/) has a [drop-in replacement](https://docs.pydantic.dev/usage/dataclasses/) for `dataclasses` and [Version 1.9.2](https://pypi.org/project/pydantic/1.9.2/) supports Python 3.6 officially. 

This would also enable us to replace [`dataclasses_json`](https://pypi.org/project/dataclasses-json/), which is not as well-supported/popular as Pydantic.

### PR DESCRIPTION
**Pull request by Navarro-Jonathan**
Replace dataclasses and dataclasses-json with the attrs library.

dataclasses are not available in python 3.6, so this is necessary for backwards compatibility.

Closes #16

### CODE REVIEW

**Comment by cartermak** [comment]
Looking through the supported Ubuntu versions for each Python version, I think we should use `ubuntu-20.04` (latest LTS that supports Python 3.6) instead of `ubuntu-latest` to get 3.6 working. Also, since we're getting specific, let's pin the version to "3.6.8" instead of just 3.6.

**Comment by Navarro-Jonathan** [comment]
> Looking through the supported Ubuntu versions for each Python version, I think we should use `ubuntu-20.04` (latest LTS that supports Python 3.6) instead of `ubuntu-latest` to get 3.6 working. Also, since we're getting specific, let's pin the version to "3.6.8" instead of just 3.6.

Sure thing. I've made those changes in daea09c

**Comment by cartermak** [comment]
Unfortunately, this version of pandas/numpy is failing to build (without some clang/C++ configuring) on my M1 Mac.

**Review by cartermak — CHANGES_REQUESTED** [review]
Mostly looks great, but comments checking on a couple of details

File: src/aerie_cli/schemas/api.py [hunk 1]
```diff
@@ -4,55 +4,70 @@
...
-@dataclass
-class ActivityBase:
+import json
+
+def convert_to_time_delta(t) -> timedelta:
+    if isinstance(t, timedelta):
+        return t
+    return postgres_interval_to_timedelta(t)
+
+def serialize_timedelta_to_postgress(inst, field, value):
```
[the remark this answers is not in the document]
Comment (Navarro-Jonathan) [reply src/aerie_cli/schemas/api.py]: This has been fixed in [c0cd0cb](https://github.com/NASA-AMMOS/aerie-cli/pull/32/commits/c0cd0cb26ef495b8c57f76ce3c4df2f9b649dddd)

File: src/aerie_cli/schemas/api.py [hunk 2]
```diff
@@ -4,55 +4,70 @@
...
-class ApiEffectiveActivityArguments:
-    arguments: dict[str, Any]
-
-
-@dataclass_json
-@dataclass
-class ActivityBase:
+import json
+
+def convert_to_time_delta(t) -> timedelta:
```
[the remark this answers is not in the document]
Comment (Navarro-Jonathan) [reply src/aerie_cli/schemas/api.py]: It is now documented in [89276b0](https://github.com/NASA-AMMOS/aerie-cli/pull/32/commits/89276b0515aa9a219958987d17610bf48db0ac9e)

File: src/aerie_cli/schemas/api.py [hunk 3]
```diff
@@ -4,55 +4,70 @@
...
+
+class ApiSerialize:
+    @classmethod
+    def from_dict(cls, dictionary: dict) -> "ApiSerialize":
+        return cls(**dictionary)
+    def to_dict(self) -> dict:
+        return asdict(self, value_serializer=serialize_timedelta_to_postgress)
+    @classmethod
+    def from_json(cls, dictionary: dict) -> "ApiSerialize":
+        return cls(**json.loads(dictionary))
```
[the remark this answers is not in the document]
Comment (Navarro-Jonathan) [reply src/aerie_cli/schemas/api.py]: Yes, they should have been Dict from typing instead. I've changed it in [f2dce67](https://github.com/NASA-AMMOS/aerie-cli/pull/32/commits/f2dce67906ac9d01ba33f90af181634379aa9ae5)

File: src/aerie_cli/schemas/api.py [hunk 4]
```diff
@@ -71,13 +85,12 @@ class ApiActivityCreate(ActivityBase):
     """
 
     plan_id: int
-    tags: list[str] = field(
-        metadata=config(encoder=lambda ts: "{" + ",".join(ts) + "}")
+    tags: List[str] = field(
+        converter = lambda ts: "{" + ",".join(ts) + "}"
```
[the remark this answers is not in the document]
Comment (Navarro-Jonathan) [reply src/aerie_cli/schemas/api.py]: That's a good point, I flipped the encoding/decoding there. It should be fixed in [de613ca](https://github.com/NASA-AMMOS/aerie-cli/pull/32/commits/de613ca2749ad0591aa139a5c3943fbd62f5b1bd)

**Comment by cartermak** [comment]
Need to pull `summary` from this query. Are integration tests downloading constraints from Aerie>1.7?)

https://github.com/NASA-AMMOS/aerie-cli/blob/5b0d373877b1fc76d42b73f7dbff0ea33eb38833/src/aerie_cli/aerie_client.py#L1631

**Review by cartermak — CHANGES_REQUESTED** [review]

**Review by cartermak — APPROVED** [review]
Nice work! Let's get this merged in and continue testing/considering any further data class changes we want to make.

**Pull request merged**